Formula/fstar.rb

```diff
@@ -16,7 +16,11 @@ class Fstar < Formula
   depends_on "opam" => :build
   depends_on "gmp"
   depends_on "ocaml" => :recommended
-  depends_on "z3" => :recommended
+
```

Comment:
would be good to add a comment about why we're using the special z3 instead of the z3 formula

Reply:
@ilovezfs Done